src/Raven.Server/Documents/BundleLoader.cs

```diff
@@ -39,18 +39,20 @@ public BundleLoader(DocumentDatabase database, ServerStore serverStore)
 
         public void HandleDatabaseRecordChange()
         {
-            TransactionOperationContext context;
-            using (_serverStore.ContextPool.AllocateOperationContext(out context))
+            lock(this)
```

Comment:
I _think_ that the internal bundles loaders are each doing their own locking.
Need to also be sure that we aren't holding this for too long (see our discussion with @karmeli87 )